crates/oxc_transformer/src/common/mod.rs

```diff
@@ -65,6 +65,16 @@ impl<'a> Traverse<'a> for Common<'a, '_> {
         self.statement_injector.exit_statements(stmts, ctx);
     }
 
+    #[inline]
+    fn enter_statement(&mut self, stmt: &mut Statement<'a>, ctx: &mut TraverseCtx<'a>) {
+        self.statement_injector.enter_statement(stmt, ctx);
+    }
+
+    #[inline]
+    fn exit_statement(&mut self, stmt: &mut Statement<'a>, ctx: &mut TraverseCtx<'a>) {
+        self.statement_injector.exit_statement(stmt, ctx);
+    }
```

Comment:
I don't think this can be correct. When exiting a statement, don't you need to restore the address of the *parent* statement? e.g.:

```js
const x = () => {
  let y;
};
```

After exiting the statement `let y`, the current statement is `const x = ...` again.

I guess would need a stack to achieve this.

Reply:
Ah, This big problem I've overlooked. Thanks for pointing it out